xc7/fasm2bels/verilog_modeling.py

```diff
@@ -114,10 +145,166 @@ def escape_verilog_name(name):
     return '\\' + name[:idx] + ' ' + name[idx:]
 
 
-class Bel(object):
-    """ Object to model a BEL.
+class ConnectionModel(object):
+    """ Constant, Wire, Bus and NoConnect objects represent a small interface
+    for Verilog module instance connection descriptions.
+    """
+
+    def to_string(self, net_map=None):
+        """ Returns the string representing this models connection in verilog.
+
+        Arguments
+        ---------
+        net_map : map of str to str
+            Optional wire renaming map.  If present, leaf wires should be
+            renamed through the map.
+
+        Returns
+        -------
+        str representing valid Verilog to model the connect by this object.
+
+        """
+        pass
+
+    def iter_wires(self):
+        """ Iterates over wires present on this object.
+
+        Yields
+        ------
+        Vector index : int
+            Is None for scalar connections, otherwise an integer that
+            represents the index into the vector.
+        Connection : str
+            Verilog representing this connection.
+
+        """
+        pass
+
+
+class Constant(ConnectionModel):
+    """ Represents a boolean constant, e.g. 1'b0 or 1'b1. """
+
+    def __init__(self, value):
+        assert value in [0, 1]
+        self.value = value
+
+    def to_string(self, net_map=None):
```

Comment:
Why `to_string` rather than `__str__`?

Reply:
Because of the net_map argument.  I don't believe arguments can be passed to __str__?